### ISSUE TITLE
SchemaGen Ant Task fails with error "schema generation failed"

### ISSUE THREAD

**Issue by rlin9** [eclipse-ee4j/jaxb-ri#1257]
java version: 10.0.2
jaxb-ri version: 2.4.0-b180830.0438

Running the sample j2s-xmlRootElement with updated classpath resulted in the following error:

```
/home/ricky/Downloads/jaxb-ri/samples/j2s-xmlRootElement/build.xml:64: schemagen failed
        at com.sun.tools.jxc.SchemaGenBase.compile(SchemaGenBase.java:260)
        at org.apache.tools.ant.taskdefs.Javac.execute(Javac.java:1133)
        at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:292)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:564)
        at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
        at org.apache.tools.ant.Task.perform(Task.java:350)
        at org.apache.tools.ant.Target.execute(Target.java:449)
        at org.apache.tools.ant.Target.performTasks(Target.java:470)
        at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1388)
        at org.apache.tools.ant.Project.executeTarget(Project.java:1361)
        at org.apache.tools.ant.helper.DefaultExecutor.executeTargets(DefaultExecutor.java:41)
        at org.apache.tools.ant.Project.executeTargets(Project.java:1251)
        at org.apache.tools.ant.Main.runBuild(Main.java:834)
        at org.apache.tools.ant.Main.startAnt(Main.java:223)
        at org.apache.tools.ant.launch.Launcher.run(Launcher.java:284)
        at org.apache.tools.ant.launch.Launcher.main(Launcher.java:101)
```

Cause seems to be the compile() method in SchemaGenBase. The "ok" variable does not change if forking is not enabled.

```
@Override
    protected void compile() {
        try {
            boolean ok = false;
            if (getFork()) {
                setupCommand();
                setupForkCommand("com.sun.tools.jxc.SchemaGeneratorFacade");
                int status = run(getCommandline().getCommandline());
                ok = (status == 0);
            } else {
                super.compile();
                //ok is still false <--------------------------
            }
            if (!ok) {
                if (!getVerbose()) {
                    log("Command invoked: " + "schemagen" + getCommandline().toString());
                }
                throw new BuildException("schemagen" + " failed", getLocation());
            }
        } catch (Exception ex) {
            if (ex instanceof BuildException) {
                throw (BuildException) ex;
            } else {
                throw new BuildException("Error starting " + "schemagen" + ": " + ex.getMessage(), ex,
                    getLocation());
            }
        }
    }
```

Thanks

**Comment by titou10titou10** [comment]
The problem exists also in v2.3.x but not in v2.2.x

**Comment by titou10titou10** [comment]
This has been fixed on the`master`branch by this commit:  c5bb550
I "guess" branch`master` is for v2.4.x
I would like to fix the bug for v2.3.x.  I can create pull request, but on what branch? v2.3.2 that has already been released?
Should we have a v2.3.3 branch? who manage this kind of things?
Thx in advance

**Comment by lukasj** [comment]
having 2.3.2 would be impossible without c5bb550, see https://github.com/eclipse-ee4j/jaxb-ri/commit/f098d53edba613f83b4b6cb3f9c330cf23f0ef01

**Comment by titou10titou10** [comment]
@lukasj , sorry, I don't get your point.
The fix is not included in v2.3.2, at least in the release present on this site 15d ago
So if I want to propose a PR to include the fix in the 2.3.x line, how do I do that? is there a v2.3.3 release planned?
Thx

**Comment by lukasj** [comment]
having 2.3.2 in staging means, that it is still subject to change at any time. If you don't see the change then the only way to get updated bits is to delete them from your local maven repo first, since maven will not re-download final versions, or build from sources from EE4J_8 branch - that is the source of truth for 2.3.2. What can be seen under 'releases' on github does not correspond to the reality yet.
